cmd/restic/cmd_rewrite.go

```diff
@@ -125,23 +131,38 @@ func rewriteSnapshot(ctx context.Context, repo *repository.Repository, sn *resti
 	}
 
 	var filter rewriteFilterFunc
+	var rewriteNode walker.NodeRewriteFunc
+	rewriteTag := "rewrite"
 
-	if len(rejectByNameFuncs) > 0 {
-		selectByName := func(nodepath string) bool {
-			for _, reject := range rejectByNameFuncs {
-				if reject(nodepath) {
-					return false
+	if opts.SnapshotSummary && sn.Summary != nil {
+		// skip snapshots which already have a summary section
+		return false, nil
+	}
+
+	if len(rejectByNameFuncs) > 0 || opts.SnapshotSummary {
+		if len(rejectByNameFuncs) > 0 {
+			selectByName := func(nodepath string) bool {
```

Comment:
`selectByName` returns true if `len(rejectByNameFuncs) == 0`. That is the `else` clause below is redundant.